### PR TITLE
add firebase sdk 3.x support

### PR DESCRIPTION
**Pull request by soumak77**
Adds basic database support to work with firebase sdk 3.x.  Many more changes needed, but this is a start.

I added a custom build with these changes to the [Firebase Template](https://github.com/base-apps/angular-firebase-template) for [Angular Base Apps](https://github.com/base-apps/angular-base-apps).  With these changes, the tests are now passing using Firebase SDK 3.x and AngularFire 2.x: 
- Source: [home-controller.js](https://github.com/base-apps/angular-firebase-template/blob/master/app/modules/home/home-controller.js)
- Tests: [home.spec.js](https://github.com/base-apps/angular-firebase-template/blob/master/app/modules/home/home.spec.js)

### CODE REVIEW

**Comment by soumak77** [comment]
I added promise support to the authentication code using [rsvp](https://www.npmjs.com/package/rsvp) which is the same library the firebase sdk appears to use.

Using a custom build with these changes, I was able to write passing unit tests for the same project specified above:
- Source: [account-controller.js](https://github.com/base-apps/angular-firebase-template/blob/master/app/modules/account/account-controller.js)
- Tests: [account.spec.js](https://github.com/base-apps/angular-firebase-template/blob/master/app/modules/account/account.spec.js)

@katowulf let me know if I'm going down the right path and I'll update the test cases

**Comment by giltotherescue** [comment]
Would love to see Firebase 3.0 support merged in. Thank you!

File: package.json [hunk 1]
```diff
@@ -52,10 +52,10 @@
     "karma-mocha": "~0.1.6",
     "karma-phantomjs-launcher": "~0.1.4",
     "karma-sinon": "~1.0.3",
+    "rsvp": "^3.2.1",
```
Comment (soumak77) [inline package.json]: @anotherpit agreed, PR has been updated

**Comment by Grantlyk** [comment]
@soumak77 Have you considered forking?

**Comment by soumak77** [comment]
@Grantlyk it's looking like I'll need to split this off into it's own repo.  Hopefully I'll find the time in the coming weeks.

**Comment by Grantlyk** [comment]
@soumak77 Your work is appreciated, an update here would be good on how you get on with that, i'd be interested to use your finished product/contribute in some way

**Comment by soumak77** [comment]
I duplicated this repo as [firebase-mock](https://github.com/soumak77/firebase-mock) and published v1.0.0 with the firebase 3.0 changes.

**Comment by prescottprue** [comment]
Thanks a ton @soumak77!

It would be nice to get these changes merged back into this repo since it is the first mock library most find.

**Comment by mkobar** [comment]
Don't know if this helps anyone but ...

I just created a [pull request #98 for firebase/angularfire-seed](https://github.com/firebase/angularfire-seed/pull/98#issuecomment-272535283) that adds firebase-3.x and angularfire-2.x support - but without the mock-firebase unit tests working.  I'm starting to look at firebase-mock to fix them all, but need to get my production system fixed first.

May help out issue #109 too.

Please let me know if anyone can help get the unit tests fixed - or even point me in the right direction.

Thanks!

**Comment by soumak77** [comment]
@mkobar you can check out this [angular template](https://github.com/base-apps/angular-firebase-template) which uses firebase-mock

**Comment by soumak77** [comment]
These changes are available in https://github.com/soumak77/firebase-mock

**Pull request closed without merging**